persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/ExpireEmbargoService.java

```diff
@@ -65,76 +61,71 @@
     public void expireEmbargoes() {
-        FedoraTransaction tx = txManager.startTransaction();
+        // get list of expired embargoes
+        List<String> resourceList = getEmbargoInfo();
+        Collection<PID> pids = new ArrayList<>();
 
-        try (Timer.Context context = timer.time()) {
-            // get list of expired embargoes
-            List<String> resourceList = getEmbargoInfo();
-            Collection<PID> pids = new ArrayList<>();
+        // remove all expired embargoes
+        for (String rescUri: resourceList) {
+            FedoraTransaction tx = txManager.startTransaction();
 
-            // remove all expired embargoes
-            for (int i = 0; i < resourceList.size(); i++) {
-                String rescUri = resourceList.get(i);
+            try (Timer.Context context = timer.time()) {
                 PID pid = PIDs.get(rescUri);
                 RepositoryObject repoObj = repoObjLoader.getRepositoryObject(pid);
-                Model model = ModelFactory.createDefaultModel().add(repoObj.getModel());
-                Resource resc = model.getResource(rescUri);
+                Resource resc = repoObj.getResource();
 
                 String eventText = null;
                 boolean expiredEmbargo = false;
-                String embargoDate = resc.getProperty(embargoUntil).getString();
 
-                if (resc.hasProperty(embargoUntil)) {
-                    repoObjFactory.deleteProperty(repoObj, embargoUntil);
-                    pids.add(pid);
-                    expiredEmbargo = true;
-                }
+                // remove embargo
+                String embargoDate = resc.getProperty(embargoUntil).getString();
+                repoObjFactory.deleteProperty(repoObj, embargoUntil);
+                pids.add(pid);
+                expiredEmbargo = true;
 
                 if (expiredEmbargo) {
-                    eventText = "Expired an embargo for " + pid.toString() + " which ended " +
+                    eventText = "Expired an embargo which ended " +
                             formatDateToUTC(parseUTCToDate(embargoDate));
-                } else {
-                    eventText = "Failed to expire embargo.";
+                    // Produce the premis event for this embargo
+                    repoObj.getPremisLog().buildEvent(Premis.Dissemination)
+                            .addSoftwareAgent(SoftwareAgent.embargoExpirationService.getFullname())
+                            .addEventDetail(eventText)
+                            .writeAndClose();
                 }
-
-                // Produce the premis event for this embargo
-                repoObj.getPremisLog().buildEvent(Premis.Dissemination)
-                        .addSoftwareAgent(SoftwareAgent.embargoExpirationService.getFullname())
-                        .addEventDetail(eventText)
-                        .writeAndClose();
+            } catch (Exception e) {
+                tx.cancelAndIgnore();
```

Comment:
You will need to log the error rather than rethrow it now so that it will attempt to continue on with the rest of the embargoed objects, otherwise the messages for successful embargo updates won't get sent.